### PR TITLE
KubeProxyConfig unit test cleanups

### PR DESCRIPTION
**Pull request by danwinship**
#### What this PR does / why we need it:
cleanups to `pkg/proxy/apis/config/validation/validation_test.go`, inspired by / taken from #121830 and #123105. A lot of this is split out of the refactoring commit from #121830, but split up more to be easier to review (and applied consistently across the whole file).

I think I'd said in #121830 that we should change the `map[string]struct{...}` in `TestValidateKubeProxyConfiguration` to `[]struct{name string, ...}` but all of the other tests also use `map[string]struct` so I just left it as-is.

The second commit ("Add names to all KubeProxyConfiguration validation test cases") is best viewed with "ignore whitespace changes".

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/kind cleanup
/sig network
/assign @aroradaman

### CODE REVIEW

File: pkg/proxy/apis/config/validation/validation_test.go [hunk 1]
```diff
@@ -37,203 +37,234 @@ func TestValidateKubeProxyConfiguration(t *testing.T) {
...
+					Min:                   ptr.To[int32](1),
+					TCPEstablishedTimeout: &metav1.Duration{Duration: 5 * time.Second},
+					TCPCloseWaitTimeout:   &metav1.Duration{Duration: 5 * time.Second},
+				},
+				Logging: logsapi.LoggingConfiguration{
+					Format: "text",
+				},
+			},
+		},
+		"dual-stack ClusterCIDR": {
```
Comment (aroradaman) [inline pkg/proxy/apis/config/validation/validation_test.go:101]: nit: "ClusterCIDR is dual stack" instead?
keeping it consistent with above "ClusterCIDR is wrong IP family".

**Review by aroradaman — COMMENTED** [review]
Really liked the output, much more readable and less pollution now.
```
=== RUN   TestValidateKubeProxyConfiguration/Invalid_number_of_ClusterCIDRs
validation_test.go:204: 
            Error Trace:    /Users/daman/github.com/aroradaman/kubernetes/pkg/proxy/apis/config/validation/validation_test.go:204
            Error:          Not equal: 
                            expected: field.ErrorList{(*field.Error)(0x140006e1a80)}
                            actual  : field.ErrorList{}
                            
                            Diff:
                            --- Expected
                            +++ Actual
                            @@ -1,8 +1,2 @@
                            -(field.ErrorList) (len=1) {
                            - (*field.Error)({
                            -  Type: (field.ErrorType) (len=17) "FieldValueInvalid",
                            -  Field: (string) (len=34) "KubeProxyConfiguration.ClusterCIDR",
                            -  BadValue: (string) (len=45) "192.168.59.0/24,fd00:192:168::/64,10.0.0.0/16",
                            -  Detail: (string) (len=88) "only one CIDR allowed or a valid DualStack CIDR (e.g. 10.100.0.0/16,fde4:8dba:82e1::/48)"
                            - })
                            +(field.ErrorList) {
                             }
            Test:           TestValidateKubeProxyConfiguration/Invalid_number_of_ClusterCIDRs
            Messages:       did not get expected validation errors
```

File: pkg/proxy/apis/config/validation/validation_test.go (hunk 1 above)
Comment (danwinship) [reply pkg/proxy/apis/config/validation/validation_test.go:101]: sure (though I copied the wording [from your commit](https://github.com/kubernetes/kubernetes/pull/121830/commits/dc7b8e8938f772b8a799dec0a7a182ab86563b5e#diff-c62251fb40adbdbf84d4b1ce8ba6c68d65996b4da9c969ec8b29e6a2cd137824R109) :slightly_smiling_face:)

Comment (danwinship) [reply pkg/proxy/apis/config/validation/validation_test.go:101]: (oh, but I guess I rewrote the one before it...)

**Pull request merged**